tests/push/test_http.py

```diff
@@ -591,7 +793,7 @@ def test_push_unread_count_message_count(self):
             self.push_attempts[5][2]["notification"]["counts"]["unread"], 4
         )
 
-    def _test_push_unread_count(self):
+    def _test_push_unread_count(self, pusher_data=None, perform_checks=True):
```

Comment:
Type annotations here, please.

That said, I'm not so keen on a method called `_test_...` which might not actually enforce any assertions, i.e. if `perform_checks` is falsey. What is the intention here? My first thought was that it was an optimisation to not run repeated checks that don't test anything new. But I think you're specifically opting out of the the checks which interrogate the unread count---because they need to be handled differently under the proposed scheme.

I think it would make more sense to create a new helper function
- which does setup only, making no assertions
- only does the minimum required to generate a counts-only and normal push notification (the present test helper seems to make many normal notifications?)